### PR TITLE
Feature: Edit ToDo Component Addition

### PR DESCRIPTION
**Pull request by ksrscharan**
## Description
- Edit Todo Component Addition.

## Database schema changes
NA

### Automated test cases added
NA

### Manual test cases run
NA

### CODE REVIEW

**Comment by ksrscharan** [comment]
Screenshot 2022-06-27 at 1 43 36 PM
Screenshot 2022-06-27 at 1 45 08 PM

File: src/components/AddTodo.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import React, { useContext } from "react";
+import { ACTION } from "../context/Actions";
+import { GlobalContext } from "../context/GlobalState";
+
+
```
Comment (sourabh-4097) [inline src/components/AddTodo.tsx]: remove extra lines

File: src/components/AddTodo.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import React, { useContext } from "react";
+import { ACTION } from "../context/Actions";
```
Comment (sourabh-4097) [inline src/components/AddTodo.tsx:3]: seperate global and local imports

File: src/components/AddTodo.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import { GlobalContext } from "../context/GlobalState";
+
+
+
+function AddTodo() {
+  const { state, dispatch } = useContext(GlobalContext); //type here
```
Comment (sourabh-4097) [inline src/components/AddTodo.tsx]: remove comment

File: src/components/EditTodo.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import React, { useContext } from "react";
+import { ACTION } from "../context/Actions";
+import { GlobalContext } from "../context/GlobalState";
+
+function EditTodo() {
+  const { state, dispatch } = useContext(GlobalContext); // type here
```
Comment (sourabh-4097) [inline src/components/EditTodo.tsx]: remove comment

File: src/context/Actions.tsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
+export type ActionType = {
+  REGISTER_TODO: string;
```
Comment (sourabh-4097) [inline src/context/Actions.tsx]: alphabetical, here and rest of the places

File: src/context/Actions.tsx [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+export const ACTION: ActionType = {
+    REGISTER_TODO: "register-todo",
+    ADD_TODO: "add-todo",
+    DELETE_TODO: "delete-todo",
+    EDIT_TODO: "edit-todo",
+    REGISTER_UPDATE: "register-update",
+    UPDATE_TITLE: "update-name",
+    CANCEL_UPDATE: "cancel-update",
+    REGISTER_DATE: "register-date",
+  };
```
Comment (sourabh-4097) [inline src/context/Actions.tsx]: add extra line

File: src/context/GlobalState.tsx [hunk 7]
```diff
@@ -0,0 +1,38 @@
+import React, {
+  createContext,
+  ReactNode,
+  useReducer,
+} from "react";
+import Reducer from "./Reducer";
+
+
```
Comment (sourabh-4097) [inline src/context/GlobalState.tsx:4]: remove extra lines

File: src/context/GlobalState.tsx [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+interface StateType{
+  todo: {}[] | {id: number; title: string; date: string}[];
+  addTodo: string;
+  isEditing: boolean;
```
Comment (sourabh-4097) [inline src/context/GlobalState.tsx:8]: alphabetical

File: src/context/GlobalState.tsx [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+const initialState: StateType | any = {
+  todo: [],
+  addTodo: "",
+  isEditing: false,
+  updatedTitle: "",
+  selectedEdit: {},
+  date: "",
+};
+
+
```
Comment (sourabh-4097) [inline src/context/GlobalState.tsx]: extra lines remove

**Comment by sourabh-4097** [comment]
Resolve merge conflict

**Pull request merged**